ecs_deploy/ecs.py

```diff
@@ -377,6 +382,124 @@ def set_commands(self, **commands):
+        self.validate_container_options(**memory)
+        for container in self.containers:
+            if container[u'name'] in memory:
+                new_memory = int(memory[container[u'name']])
+                diff = EcsTaskDefinitionDiff(
+                    container=container[u'name'],
+                    field=u'memory',
+                    value=new_memory,
+                    old_value=container.get(u'memory')
+                )
+                self._diff.append(diff)
+                container[u'memory'] = new_memory
+    
+    def set_memoryreservation(self, **memoryreservation):
+        self.validate_container_options(**memoryreservation)
+        for container in self.containers:
+            if container[u'name'] in memoryreservation:
+                new_memoryreservation = int(memoryreservation[container[u'name']])
+                diff = EcsTaskDefinitionDiff(
+                    container=container[u'name'],
+                    field=u'memoryReservation',
+                    value=new_memoryreservation,
+                    old_value=container.get(u'memoryReservation')
+                )
+                self._diff.append(diff)
+                container[u'memoryReservation'] = new_memoryreservation
+
+    def set_privileged(self, **privileged):
+        self.validate_container_options(**privileged)
+        for container in self.containers:
+            if container[u'name'] in privileged:
+                new_privileged = bool(privileged[container[u'name']])
+                old_privileged = container.get(u'privileged')
+                if not new_privileged == old_privileged: 
+                    diff = EcsTaskDefinitionDiff(
+                        container=container[u'name'],
+                        field=u'privileged',
+                        value=new_privileged,
+                        old_value=old_privileged
+                    )
+                    self._diff.append(diff)
+                    container[u'privileged'] = new_privileged
+
+    def set_essential(self, **essential):
+        self.validate_container_options(**essential)
+        for container in self.containers:
+            if container[u'name'] in essential:
+                new_essential = bool(essential[container[u'name']])
+                old_essential = container.get(u'essential')
+                if not new_essential == old_essential: 
+                    diff = EcsTaskDefinitionDiff(
+                        container=container[u'name'],
+                        field=u'essential',
+                        value=new_essential,
+                        old_value=old_essential
+                    )
+                    self._diff.append(diff)
+                    container[u'essential'] = new_essential
+
+    def set_log_configurations(self, log_configurations_list):
+        log_configurations = {}
+        for log_configuration in log_configurations_list:
+            log_configurations.setdefault(log_configuration[0], {})
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `dict.setdefault()` method. We recommend using the `defaultdict` object to handle missing items in internal state because the default value is automatically generated when you access or modify a missing key. `defaultdict` can also be easier to use and understand.

[Learn more](https://docs.python.org/3/library/collections.html#collections.defaultdict)

Reply:
Should be fine now.